File: src/main/java/com/ebuozturk/ecommerce/dto/ProductDto.java
package com.ebuozturk.ecommerce.dto;


import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class ProductDto {

    private Long id;
    private String name;
    private Double unitPrice;
    private Integer quantityPerUnit;
    private Integer unitsInStock;
    private CategoryDto category;
}

File: src/main/java/com/ebuozturk/ecommerce/dto/product/ProductDto.kt
package com.ebuozturk.ecommerce.dto.product;

import com.ebuozturk.ecommerce.dto.category.CategoryDto

data class ProductDto @JvmOverloads constructor(
    val id: String? = "",
    val name: String,
    val unitPrice: Double,
    val quantityPerUnit: Int,
    val unitsInStock: Int ,
    val category: CategoryDto
){

}

